src/library/coverartcache.cpp

```diff
@@ -89,16 +89,18 @@ QPixmap CoverArtCache::tryLoadCover(
     DEBUG_ASSERT(!pTrack ||
                 pTrack->getLocation() == coverInfo.trackLocation);
 
+    const auto requestedImageHash = coverInfo.imageHash();
+
     if (coverInfo.type == CoverInfo::NONE) {
         if (loading == Loading::Default) {
-            emit coverFound(pRequestor, coverInfo, QPixmap(), coverInfo.hash, false);
+            emit coverFound(pRequestor, coverInfo, QPixmap(), requestedImageHash, false);
         }
         return QPixmap();
     }
 
     // keep a list of trackIds for which a future is currently running
     // to avoid loading the same picture again while we are loading it
-    QPair<const QObject*, quint16> requestId = qMakePair(pRequestor, coverInfo.hash);
+    QPair<const QObject*, mixxx::cache_key_t> requestId = qMakePair(pRequestor, requestedImageHash);
```

Comment:
I think @daschuer mentioned that we should avoid QPair, but I don't remember why.

Reply:
I don't see any value in refactoring this. In Rust we often use *tuple structs* as long as the anonymous members are clearly distinguishable by their type. Even C++ introduced better handling of tuples and binding to variables with `std::tie`.

Reply:
Ok :D